mlflow/pipelines/steps/train.py

```diff
@@ -158,6 +171,69 @@ def _run(self, output_directory):
             raw_validation_df, prediction_result, self.target_col
         )
 
+        from mlflow.entities import ViewType
+
+        mlflow_client = MlflowClient()
+        exp_id = _get_experiment_id()
+        search_result = mlflow_client.search_runs(
+            experiment_ids=exp_id,
+            run_view_type=ViewType.ALL,
+            max_results=10000,
+            order_by=["attribute.start_time DESC", f"metrics.{self.primary_metric}"],
+        )
+
+        primary_metric_name = _get_primary_metric(self.step_config)
+        custom_metric_names = [cm["name"] for cm in _get_custom_metrics(self.step_config)]
+        metric_greater_is_better = _get_metric_greater_is_better(self.step_config)
+        metric_names = metric_greater_is_better.keys()
+
+        metric_keys = [f"{metric_name}_on_data_validation" for metric_name in metric_names]
+
+        leader_board_items = []
+        for old_run in search_result:
+            if all(metric_key in old_run.data.metrics for metric_key in metric_keys):
+                metric_values = {
+                    metric_name: "{:.6g}".format(old_run.data.metrics[metric_key])
+                    for metric_name, metric_key in zip(metric_names, metric_keys)
+                }
+                leader_board_items.append(
+                    {
+                        "Run ID": old_run.info.run_id,
+                        "Run Time": datetime.datetime.fromtimestamp(
+                            old_run.info.start_time // 1000
+                        ),
+                        **metric_values,
+                    }
+                )
+
+        leader_board_items.sort(
+            key=lambda x: x[primary_metric_name],
+            reverse=metric_greater_is_better[primary_metric_name],
+        )
+        for i, leader_board_item in enumerate(leader_board_items):
+            leader_board_item["Model Rank"] = f"{i + 1} / {len(leader_board_items)}"
+        latest_model_index = list(map(lambda x: x["Run ID"], leader_board_items)).index(
+            run.info.run_id
+        )
+        latest_model_item = leader_board_items[latest_model_index]
+
+        # metric columns order: primary metric, then custom metrics, then builtin metrics.
+        metric_columns = (
+            [primary_metric_name]
+            + sorted(list(set(custom_metric_names) - {primary_metric_name}))
+            + sorted(list(set(metric_names) - set(custom_metric_names) - {primary_metric_name}))
+        )
```

Comment:
```suggestion
        # metric columns order: primary metric, then custom metrics, then builtin metrics.
        def sorter(m):
            if m == primary_metric_name:
                return 0
            elif m in custom_metric_names:
                return 1
            else:
                return 2
         
         metric_columns = sorted(metric_names, key=sorter)
```

Reply:
Not sure if the suggested approach will work within each category by sorting them alphabetically.

Reply:
If we want to sort the items within each category, we can return `(num, m)` in `sorter`.